crates/storage/storage.rs

```diff
@@ -247,6 +247,9 @@ impl Store {
         }
 
         // Store chain info
+        if let Some(cancun_time) = genesis.config.cancun_time {
+            self.update_cancun_time(cancun_time)?;
```

Comment:
these values are set once. Wouldn't it make more sense to have a function like `set_chain_spec` that sets these values the first time? All these values shouldn't be updatable.

Reply:
sure

Reply:
Updated!